client/webui/frontend/src/lib/components/chat/ConnectionRequiredModal.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * Connection Required Modal Component
+ *
+ * Shows when user tries to select an enterprise source that requires authentication
+ * Provides option to navigate to connections panel to authenticate
+ */
+
+import React from "react";
+import { AlertCircle, ExternalLink } from "lucide-react";
+import { Button } from "@/lib/components/ui/button";
+
+interface ConnectionRequiredModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    sourceName: string;
+    onNavigateToConnections: () => void;
+}
+
+export const ConnectionRequiredModal: React.FC<ConnectionRequiredModalProps> = ({ isOpen, onClose, sourceName, onNavigateToConnections }) => {
```

Comment:
Is this a new dialog component? If so, we already have one, so unless there is a justifiable need for creating a new one, let's please use the existing component instead.